tests/serverpod_test_server/test_integration/session/validate_headers_config_test.dart

```diff
@@ -0,0 +1,234 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/src/web/routes/session_test_route.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+// Custom auth provider that returns a raw unwrapped token
+class _UnwrappedTokenAuthProvider implements ClientAuthKeyProvider {
+  final String token;
+
+  _UnwrappedTokenAuthProvider(this.token);
+
+  @override
+  Future<String?> get authHeaderValue async => token;
+}
+
+// Custom auth provider that returns a Bearer-wrapped token
+class _BearerTokenAuthProvider implements ClientAuthKeyProvider {
+  final String token;
+
+  _BearerTokenAuthProvider(this.token);
+
+  @override
+  Future<String?> get authHeaderValue async => 'Bearer $token';
+}
+
+// Custom auth provider that returns a Basic-wrapped token
+class _BasicTokenAuthProvider implements ClientAuthKeyProvider {
+  final String token;
+
+  _BasicTokenAuthProvider(this.token);
+
+  @override
+  Future<String?> get authHeaderValue async =>
+      'Basic ${base64.encode(utf8.encode(token))}';
+}
+
+void main() {
+  group('Given a server with validateHeaders enabled', () {
+    late Serverpod server;
+    late Client client;
+    const testToken = 'test-token';
+
+    setUpAll(() async {
+      server = IntegrationTestServer.create(
+        authenticationHandler: (final session, final token) async {
+          if (token == testToken) {
+            return AuthenticationInfo(
+              'test-user-123',
```

Comment:
For repeated test strings, extract them into `const` variables and reuse them across the tests